### PR TITLE
Adding tick marks functionality

### PR DESCRIPTION
**Pull request by octavianCraciun**
Tick marks support the following cases:
- Major Ticks Marks visible (showTicks : true, showMinorTicks : false)
- Major & Minor Ticks Marks visible (showTicks : true, showMinorTicks : true)
- All Tick Marks hidden (showTicks : false)

Add missing properties for grid lines

### CODE REVIEW

File: jquery.flot.js [hunk 1]
```diff
@@ -1670,40 +1693,94 @@ Licensed under the MIT license.
             x = edges.x;
             y = edges.y;
 
-            // draw major tick marks
+			// draw major tick marks
```
Comment (cristiingineru) [inline jquery.flot.js]: You have a few tabs here.

File: jquery.flot.js [hunk 2]
```diff
@@ -1670,40 +1693,94 @@ Licensed under the MIT license.
...
             for (i = 0; i < axis.ticks.length; ++i) {
                 var v = axis.ticks[i].v,
                     xoff = 0,
-                    yoff = 0;
+                    yoff = 0;           
 
-                if (isNaN(v) || v < axis.min || v > axis.max)
-                    continue;
+                if (!isNaN(v) && v >= axis.min && v <= axis.max) {       
+                    if (axis.direction == "x") {
```
Comment (cristiingineru) [inline jquery.flot.js]: Use === and !== for a faster execution.

File: jquery.flot.js [hunk 3]
```diff
@@ -1670,40 +1693,94 @@ Licensed under the MIT license.
...
+                        xminor = 0,
+                        yminor = 0,
+                        j;
+
+                    for(j = 1; j <= minorTicksNr; j++)
+                    {
+                        // compute minor tick position
+                        if(axis.direction == "x")
+                        {
+                            yminor = t / 2; // minor ticks are half length
```
Comment (cristiingineru) [inline jquery.flot.js]: Move these "constant" initialization outside the for loop.

File: jquery.flot.js [hunk 4]
```diff
@@ -1670,40 +1693,94 @@ Licensed under the MIT license.
...
-                        xoff = -xoff;
-                }
+                    ctx.moveTo(x, y);
+                    ctx.lineTo(x + xoff, y + yoff);
+                }
+
+                //draw minor tick marks
+                if(minorTicks == true && i < axis.ticks.length - 1)
+                {
+                    var minorTicksNr = 4; // default number of minor ticks
```
Comment (cristiingineru) [inline jquery.flot.js]: Move this constant and the default tick length to jquery.flot.uiConstants.js file

File: jquery.flot.js [hunk 5]
```diff
@@ -1030,17 +1036,30 @@ Licensed under the MIT license.
...
+                showMinorTicks = true;
+            }
+
+            // By default, grid lines are visible
+            if (gridLines == null) {
+                gridLines = true;
+            }
+
             if (!isNaN(+tickLength))
                 padding += +tickLength;
```
Comment (cristiingineru) [inline jquery.flot.js]: Not directly related to your changes, but are these lines working fine? Those 2 double tickLength increments look suspicious.

File: jquery.flot.js [hunk 6]
```diff
@@ -1670,40 +1693,94 @@ Licensed under the MIT license.
             x = edges.x;
             y = edges.y;
```
Comment (cristiingineru) [inline jquery.flot.js]: Feel free to move these initializations in the var section.

File: jquery.flot.js [hunk 7]
```diff
@@ -1670,40 +1693,94 @@ Licensed under the MIT license.
...
-                ctx.moveTo(x, y);
-                ctx.lineTo(x + xoff, y + yoff);
+                            // don't go over the plot borders
+                            if((x < 0) || (x > plotWidth))
+                                continue;
+                        } else {
+                            xminor = t / 2; // minor ticks are half length
+                            y = axis.p2c(v1 + j*step);
+
+                            if (ctx.lineWidth == 1) {
```
Comment (cristiingineru) [inline jquery.flot.js]: Could you please define a local function variable here that does this correction and reused it for both x and y? Something like:
var crispy = function(d) {
  return ctx.lineWidth === 1 ? Math.floor(d) + 0.5 : d;
}

Comment (cipix2000) [inline jquery.flot.js]: in fact there are two cases, odd line width and even line width, not just 1

**Comment by octavianCraciun** [comment]
I changed the examples to work again(added the uiConstants.js as dependency)

**Pull request merged**